### PR TITLE
Feat: Invoice collection account selector page

### PR DESCRIPTION
**Pull request by hungvu193**
### Fixed Issues

$ https://github.com/Expensify/App/issues/39752
PROPOSAL: N/A

### Tests

Precondition: User connected to Xero and had bank accounts.
1. Go to Workspace => Accounting => Advance => Xero Invoice Collections Account
2. Click any bank account
3. Verify that it's selected
4. Verify It displays the correct bank account name under `Xero Invoice Collections Account` in Advance page.
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests.

### QA Steps

Same as Tests.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/16502320/79070bd8-59d0-4a93-a3fc-e6b9ee0a4d91

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2024-05-03 at 23 23 23

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/16502320/f66a1f6e-6ecf-488d-b9ee-73c5e995b681

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/16502320/367f7e03-f0e0-4373-961a-a0988b25d297

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/16502320/373ead5b-c49e-4505-a9af-cbbf84934b10

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/16502320/742ecbc3-0d07-4a60-9ae7-9bac68ee1b6b

</details>

### CODE REVIEW

**Comment by rushatgabhane** [comment]
Hold can be lifted. PR merged

**Comment by rushatgabhane** [comment]
please merge latest main

**Comment by hungvu193** [comment]
> please merge latest main

Sure things. I'm finishing `SelectionScreen` before merging with main

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

File: src/components/SelectionScreen.tsx
Comment (hungvu193) [reply src/components/SelectionScreen.tsx:51]: Yeah it will cause padding issue cause we're using SelectionList.

**Comment by hungvu193** [comment]
Will be ready for another review @rushatgabhane

**Comment by rojiphil** [comment]
Are there two C+ reviewers required here? If not, please remove me as a reviewer here.

**Comment by hungvu193** [comment]
> Are there two C+ reviewers required here? If not, please remove me as a reviewer here.

Ah , thanks @rojiphil. @rushatgabhane will handle it

File: src/components/SelectionScreen.tsx
Comment (mananjadhav) [inline src/components/SelectionScreen.tsx:10]: Can we use absolute import?

File: src/components/SelectionScreen.tsx
Comment (hungvu193) [reply src/components/SelectionScreen.tsx:1]: I'm not sure, do have any documentation about when should we create storybook?

File: src/components/SelectionScreen.tsx
Comment (mananjadhav) [inline src/components/SelectionScreen.tsx:19]: Can we add newline between in each prop for readability?

File: src/components/SelectionScreen.tsx
Comment (mananjadhav) [reply src/components/SelectionScreen.tsx:59]: Considering we already have the PR open for this one can we now use it? Avoid unnecessary effort later to refactor.

File: src/components/SelectionScreen.tsx
Comment (mananjadhav) [reply src/components/SelectionScreen.tsx:10]: thanks for clarifying.

File: src/languages/en.ts
Comment (mananjadhav) [inline src/languages/en.ts:2013]: We can ask someone to verify the copy.

```suggestion
                invoiceAccountSelectDescription: "As you've enabled exporting invoices from Expensify to Xero, this is the account the invoice will appear against once marked as paid.",
```

File: src/pages/workspace/accounting/xero/advanced/XeroAdvancedPage.tsx
Comment (mananjadhav) [inline src/pages/workspace/accounting/xero/advanced/XeroAdvancedPage.tsx:27]: Does this have to be inside `useMemo`? Can we take this outside and add `invoiceCollectionsAccountID` as dependency?

File: src/languages/en.ts
Comment (hungvu193) [reply src/languages/en.ts:2013]: hmm I copied this from docs, let me check again

Comment (hungvu193) [reply src/languages/en.ts:2013]: I think current copy is correct

Screenshot 2024-05-03 at 22 29 22

File: src/pages/workspace/accounting/xero/advanced/XeroAdvancedPage.tsx
Comment (hungvu193) [reply src/pages/workspace/accounting/xero/advanced/XeroAdvancedPage.tsx]: updated!

File: src/languages/en.ts
Comment (mananjadhav) [reply src/languages/en.ts:2013]: I agree it's correct when compared to the document, but at Expensify we generally follow sentence case. 

cc - @trjExpensify can chime in.

File: src/components/SelectionScreen.tsx
Comment (hungvu193) [reply src/components/SelectionScreen.tsx:69]: Unfortunately, we're still having discussion (existing PR) about `ConnectionLayout`. I think we still should keep the code like it's right now.

File: src/components/SelectionScreen.tsx
Comment (hungvu193) [reply src/components/SelectionScreen.tsx:1]: I feel like only primitive components like Input, Form will need storybook. But can you raise slack post please?

File: src/languages/es.ts
Comment (mananjadhav) [inline src/languages/es.ts:2047]: nit, for brewity should we name the key as 

```suggestion
                invoiceAccountSelectorDescription:
```

File: src/pages/workspace/accounting/xero/advanced/XeroInvoiceAccountSelectorPage.tsx
Comment (mananjadhav) [inline src/pages/workspace/accounting/xero/advanced/XeroInvoiceAccountSelectorPage.tsx:68]: When do hit the URL in the address bar, this will go back directly to accounting. Are we okay with that?

**Comment by mananjadhav** [comment]
@hungvu193 The screenshots are missing for all platforms, can you add those please?

File: src/components/SelectionScreen.tsx
Comment (mananjadhav) [reply src/components/SelectionScreen.tsx:1]: Raised it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1714751862924539). IMO this shouldn't block this PR though.

File: src/languages/en.ts
Comment (trjExpensify) [reply src/languages/en.ts:2013]: Yeah, agreed with Manan!

File: src/pages/workspace/accounting/xero/advanced/XeroInvoiceAccountSelectorPage.tsx
Comment (hungvu193) [reply src/pages/workspace/accounting/xero/advanced/XeroInvoiceAccountSelectorPage.tsx]: nice catch! Updated

**Comment by hungvu193** [comment]
I addressed the comments and added videos @mananjadhav

**Comment by mananjadhav** [comment]
Thanks the code changes look fine, will work on the checklist.

**Comment by mananjadhav** [comment]
@hungvu193 I see two problems when testing.

1. This view should only be visible if Sync reimbursed reports is enabled. While the menu item visibility works fine, `https://dev.new.expensify.com:8082/settings/workspaces/76A10AB364F95DA0/accounting/xero/advanced/invoice-account-selector` if I hit the URL with the sync disabled, I can still see the page and able to update the setting as well. 

2. I am not sure if it's a navigation issue/related to this PR, but at times I don't see `/invoice-account-selector` in the URL. The page loads on click but ends at `/xero/advanced` or `/import` if I have navigated to import earlier.


Screenshot 2024-05-04 at 11 23 50 PM

**Comment by hungvu193** [comment]
> @hungvu193 I see two problems when testing.
> 
> 1. This view should only be visible if Sync reimbursed reports is enabled. While the menu item visibility works fine, `https://dev.new.expensify.com:8082/settings/workspaces/76A10AB364F95DA0/accounting/xero/advanced/invoice-account-selector` if I hit the URL with the sync disabled, I can still see the page and able to update the setting as well.
> 2. I am not sure if it's a navigation issue/related to this PR, but at times I don't see `/invoice-account-selector` in the URL. The page loads on click but ends at `/xero/advanced` or `/import` if I have navigated to import earlier.
> 
> Screenshot 2024-05-04 at 11 23 50 PM

I can't reproduce the `2`. With the first one, we need to implement  a new props for our `AccessOrNotFoundWrapperProps`, currently it's only showing NotFoundPage bases on `accessVariants` and `featureName`.

**Comment by mananjadhav** [comment]
About 2, even I don't have proper reproduction steps. For 1, do we have any workaround? I implemented it by not passing the policyID in the categories page. But we can let @lakchote @mountiny confirm if we need it in this PR. I'll continue with the checklist in the current state.

**Comment by hungvu193** [comment]
> For 1, do we have any workaround?

I'm trying to update `AccessOrNotFoundWrapper` but I'm facing the hard issue "naming the variable" 😂

**Comment by mananjadhav** [comment]
> > For 1, do we have any workaround?
> 
> I'm trying to update `AccessOrNotFoundWrapper` but I'm facing the hard issue "naming the variable" 😂

But are you like adding a boolean prop, which will show the NotFound if it's true? along with feature names and variants? We'll need it to be generic as the conditions would be different for some of the pages.

**Comment by hungvu193** [comment]
> > > For 1, do we have any workaround?
> > 
> > 
> > I'm trying to update `AccessOrNotFoundWrapper` but I'm facing the hard issue "naming the variable" 😂
> 
> But are you like adding a boolean prop, which will show the NotFound if it's true? along with feature names and variants? We'll need it to be generic as the conditions would be different for some of the pages.

Yeah, I want to add a new boolean props, ie: `shouldBeBlocked` and update this condition:
```javascript
const shouldShowNotFoundPage = isEmptyObject(policy) || (Object.keys(policy).length === 1 && !isEmptyObject(policy.errors)) || !policy?.id || !isPageAccessible || !isFeatureEnabled;
```
to

```javascript
const shouldShowNotFoundPage = isEmptyObject(policy) || (Object.keys(policy).length === 1 && !isEmptyObject(policy.errors)) || !policy?.id || !isPageAccessible || !isFeatureEnabled || shouldBeBlocked;
```

**Comment by mananjadhav** [comment]
I think `shouldBeBlocked` should be okay as `showShowNotFound` is already used in the component and it would lead to confusion.

**Comment by hungvu193** [comment]
OK, let me update it then

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/60b6851c-c0c7-460a-bc38-5dfa1eed202e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/e415cd95-1ede-409a-98e3-151166281176

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/f1414318-7dde-41b0-b8c5-20ca0f17ca82

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/71bb965d-79a1-43dd-8e1f-83a9428f7888

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3069065/5c2dce82-3df6-455d-b0eb-8a42137e24f6

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/43ead238-e0eb-4d34-a0f1-dd32f97ab62d

</details>

**Comment by hungvu193** [comment]
It should be working now.

https://github.com/Expensify/App/assets/16502320/8cfe7911-57fd-4e3a-a37d-b7350b6bac5d

**Comment by mananjadhav** [comment]
Tested the block access change. Added the screencast for Web.

**Review by mananjadhav — APPROVED** [review]

**Review by mananjadhav — APPROVED** [review]

**Review by lakchote — APPROVED** [review]

**Comment by lakchote** [comment]
Unassigning other reviewers as reviews have already been done by 2 C+ contributors.

**Comment by shawnborton** [comment]
Design-wise I think this is looking good.

**Pull request merged**
